packages/@apphosting/adapter-angular/src/bin/build.ts

```diff
@@ -1,20 +1,87 @@
 
+interface OutputBundleOptions {
+  bundleYamlPath: string;
+  outputDirectory: string;
+  outputBaseDirectory: string;
+  serverFilePath: string;
+  browserDirectory: string;
+}
+
+const cwd = process.cwd();
+
+// read Application Builder config
+const config: ApplicationBuilderOptions = await loadConfig(cwd);
+const outputPath = config.outputPath;
+// normalized output path structure
+const normalizedOutputPath = {
+  browser: "browser",
+  server: "server",
+  media: "media",
+  ...(typeof outputPath === "string" ? undefined : outputPath),
+  base: normalize(resolve(typeof outputPath === "string" ? outputPath : outputPath.base)),
+};
+
+// populate file/directory paths we need inside app hosting output directory
+function populateOutputBundleOptions(): OutputBundleOptions {
+  const outputBundleDir = resolve(".apphosting");
+  return {
+    bundleYamlPath: resolve(outputBundleDir, "bundle.yaml"),
+    outputDirectory: outputBundleDir,
+    outputBaseDirectory: resolve(outputBundleDir, "dist"),
+    serverFilePath: resolve(outputBundleDir, "dist", normalizedOutputPath.server, "server.mjs"),
+    browserDirectory: resolve(outputBundleDir, "dist", normalizedOutputPath.browser),
+  };
+}
+
+// Run build command
 const build = (cwd = process.cwd()) =>
   new Promise<void>((resolve, reject) => {
-    // TODO warn if the build script contains anything other than `ng build`
     const process = spawn("npm", ["run", "build"], { cwd, shell: true, stdio: "inherit" });
     process.on("exit", (code) => {
       if (code === 0) return resolve();
       reject();
     });
   });
 
-const config = await loadConfig(process.cwd());
+// move the base output directory, which contains the server and browser bundle directory, and prerendered routes
+// as well as generating bundle.yaml
+async function generateOutputDirectory(
+  cwd: string,
+  outputBundleOptions: OutputBundleOptions,
+): Promise<void> {
+  const baseDirectory = resolve(normalizedOutputPath.base);
 
-await build().catch(() => process.exit(1));
+  await Promise.all([
+    move(baseDirectory, outputBundleOptions.outputBaseDirectory, { overwrite: true }),
+    generateBundleYaml(outputBundleOptions, cwd),
+  ]);
+  return;
+}
+
+// generate bundle.yaml
+async function generateBundleYaml(
+  outputBundleOptions: OutputBundleOptions,
+  cwd: string,
+): Promise<void> {
+  await writeFile(
+    outputBundleOptions.bundleYamlPath,
+    yamlStringify({
+      headers: [],
+      redirects: [],
+      rewrites: [],
+      runCommand: `node ${normalize(relative(cwd, outputBundleOptions.serverFilePath))}`,
+      neededDirs: [normalize(relative(cwd, outputBundleOptions.outputDirectory))],
+      staticAssets: [normalize(relative(cwd, outputBundleOptions.browserDirectory))],
+    }),
+  );
+  return;
+}
 
-// TODO do all the things
-console.log({ config });
+const outputBundleOptions = populateOutputBundleOptions();
+await build().catch(() => process.exit(1));
```

Comment:
Whats the intention of catching and then doing process.exit?

Reply:
The 'spawn' event will fire regardless of whether an error occurs within the spawned process. So in case reject(), we can catch and exit with 1